dtonator/src/main/java/com/bizo/dtonator/properties/ReflectionTypeOracle.java

```diff
@@ -23,7 +23,9 @@
         pd.getReadMethod() == null ? pd.getPropertyType().getName() : pd.getReadMethod().getGenericReturnType().toString().replaceAll("^class ", ""),
         pd.getWriteMethod() == null,
         pd.getReadMethod() == null ? null : pd.getReadMethod().getName(),
-        pd.getWriteMethod() == null ? null : pd.getWriteMethod().getName()));
+        pd.getWriteMethod() == null ? null : pd.getWriteMethod().getName(),
```

Comment:
I'm curious, if instead of tracking these declaring class properties, only to ignore them later, can we just not return them from `ReflectionTypeOracle.getProperties` at all?

E.g. right now `getProperties` returns `Props` in both the class + all sub-classes.

If we change that to only return `Props` in just the class itself, would that solve the problem of duplicate `Props` in a potentially cleaner way?

Reply:
If I understand correctly, you are suggesting doing the filtering in the ReflectionTypeOracle.getProperties() call rather than in DtoConfig right? That could work, but it would require passing in an additional parameter to tell the oracle getProperties() method whether or not the parent class properties should be filtered. I can give this a try and see if I run into any difficulties.